hls-plugin-api/hls-plugin-api.cabal

```diff
@@ -57,6 +58,7 @@ library
     , transformers
     , unordered-containers
     , megaparsec > 9
+    , hw-fingertree
```

Comment:
I'm not keen on adding new dependencies to `his-plugin-api`. Other alternatives worth considering:

- Extract `RangeMap` to a standalone package 
- Leave it in its current home - plugins can depend on each other just fine

Reply:
Although I think the functionality is pretty small to warrant a standalone package, I think I can do that. I am not too keen on the second option. Even if the plugins _can_ depend on each other, I think having this as part of a specific plugin signals the idea that this is something specific to that plugin. I think that may discourage plugin authors to use it (it certainly would discourage me).

What is the downside of adding new dependencies to `hls-plugin-api`?

Reply:
> What is the downside of adding new dependencies to hls-plugin-api?

When we update HLS for a new version of GHC, `hls-plugin-api` is one of the packages that needs to get updated if we are to do _anything_. We are frequently blocked by upstream dependencies not working with the new version of GHC; every new dependency that we add to key packages is another opportunity for that to happen...

Reply:
That said, `hw-fingertree` depends only on `base`, which is promising.

Reply:
I see. Indeed, I had to add GHC 9.4 support to `hw-fingertree` myself, so I can definitely see that happening :sweat_smile: Then I will extract this into a separate small package.

Reply:
> Personally I suggest extracting it to a _module_ in `hls-plugin-api` or something.

I am a bit confused by this. Do you suggest something other than what currently is present in the PR? Because it currently _is_ a module within `hls-plugin-api` (unless there is a separate concept of _module_ which I'm not aware).

> what if we add a cabal flag to `hls-plugin-api` `use-fingertree`

Sounds good to me. What do you think @pepeiborra?

Reply:
Ah, I think I misunderstood. I thought we were talking about extracting the _fingertree_ code from `hw-fingertree` so we controlled it, rather than extracting `RangeMap`. Ignore me, then.

> what if we add a cabal flag to hls-plugin-api use-fingertree

... but I still think this is the cleanest solution.

Reply:
I have put the fingertree implementation behind a flag. All related tests (tests of `RangeMap` itself and the plugins that make use of it) pass with both implementations. Can you take another look to see if it reflects what you had in mind? @michaelpj